src/qbox_statemachine_demo.py

```diff
@@ -0,0 +1,219 @@
+    finish_state = 4
+
+class StateMachine:
+    def __init__(self, config):
+        self.config = config
+        self.transaction_number = len(config["onrequest"]["transactions"])
+        self.state_info = StateInfo(config, [StateInfo.INIT for i in range(self.transaction_number)])
+        self.state_type = State.start_state
+        self.response = [None] * self.transaction_number
+        self.saga_fail = False
+
+    def transition(self):
+        if self.state_type == State.start_state:
+            res = send_requests(config)    
+            if res:
+                for i in range(self.transaction_number):
+                    self.state_info.change_state(i, StateInfo.AWAIT)
+                self.state_type = State.awaiting_state
+        elif self.state_type == State.awaiting_state:
+            srcname, info = get_respond()
+            for i, transaction in enumerate(self.config["onrequest"]["transactions"]):
+                if transaction["dstname"] == srcname:
+                    if info == transaction["success"]:
+                        if self.state_info.get_state(i) == StateInfo.AWAIT:
+                            self.state_info.change_state(i, StateInfo.SUCCESS)
+                        else:
+                            print("somethin weird")
+                    elif info == transaction["fail"] or transaction["fail"] == QboxConfig.FAIL_REST:
+                        if self.state_info.get_state(i) == StateInfo.AWAIT:
+                            self.state_info.change_state(i, StateInfo.FAIL)
+                        else:
+                            print("somethin weird")
+            if self.state_info.get_all_response():
+                res = self.state_info.get_response_status()
+                if res in [StateInfo.ALL_SUCCESS, StateInfo.ALL_FAIL]:
+                    self.state_type = State.finish_state
+                    if res == StateInfo.ALL_FAIL:
+                        self.saga_fail = True
+                else:
+                    self.state_type = State.compensation_state
+        
+        elif self.state_type == State.compensation_state:
+            self.saga_fail = True
+            all_compensated = True
+            for i, transaction in enumerate(self.config["onrequest"]["transactions"]):
+                if self.state_info.get_state(i) == StateInfo.FAIL:
+                    res = send_compensation(self.config)
+                    if res:
+                        self.state_info.change_state(i, StateInfo.COMPENSATED)
+                    else:
+                        all_compensated = False
+            if all_compensated:
+                self.state_type = State.finish_state
+        elif self.state_type == State.finish_state:
+            res = send_back_result_to_requestor(self.config, self.saga_fail)
+            if res:
+                return True
+        return False
+
+        
+if __name__ == "__main__":
+    
+    sm = StateMachine(config)
+    print(sm.state_info)

```

Comment:
Use `logging.info` not `print` so that we can disable and configure accordingly.